inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_pooling_node.cpp

```diff
@@ -372,14 +446,15 @@ void MKLDNNROIPoolingNode::createPrimitive() {
         roi_pooling_kernel->create_ker();
 }
 
-void MKLDNNROIPoolingNode::execute(mkldnn::stream strm) {
+template<typename T>
+void MKLDNNROIPoolingNode::execute() {
     auto &srcMemory0 = getParentEdgeAt(0)->getMemory();
     auto &srcMemory1 = getParentEdgeAt(1)->getMemory();
-    auto &dstMemory = getChildEdgeAt(0)->getMemory();
+    auto &dstMemory  = getChildEdgeAt(0)->getMemory();
 
-    const auto *src_data = reinterpret_cast<const float *>(srcMemory0.GetPtr());
-    const auto *src_roi = reinterpret_cast<const float *>(srcMemory1.GetPtr());
-    float *dst = reinterpret_cast<float *>(dstMemory.GetPtr());
+    const auto *src_data = reinterpret_cast<const T*>(srcMemory0.GetPtr());
+    const auto *src_roi  = reinterpret_cast<const T*>(srcMemory1.GetPtr());
+    auto       *dst      = reinterpret_cast<T*>(dstMemory.GetPtr());
```

Comment:
NIT. Just use `auto` only since the type is completely described in the `reinterpret_cast` template argument.